commit b3449922502f5a161ee2b5022a33aec8472fbf18
Author: Oleg Nesterov <oleg@redhat.com>
Date:   Fri Mar 23 15:02:47 2012 -0700

    usermodehelper: introduce umh_complete(sub_info)
    
    Preparation.  Add the new trivial helper, umh_complete().  Currently it
    simply does complete(sub_info->complete).
    
    Signed-off-by: Oleg Nesterov <oleg@redhat.com>
    Cc: Tetsuo Handa <penguin-kernel@I-love.SAKURA.ne.jp>
    Cc: Rusty Russell <rusty@rustcorp.com.au>
    Cc: Tejun Heo <tj@kernel.org>
    Cc: David Rientjes <rientjes@google.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
    [dannf: Adjusted to apply to Debian's 2.6.32]

diff --git a/kernel/kmod.c b/kernel/kmod.c
index d206078..c6059c4 100644
--- a/kernel/kmod.c
+++ b/kernel/kmod.c
@@ -203,6 +203,11 @@ void call_usermodehelper_freeinfo(struct subprocess_info *info)
 }
 EXPORT_SYMBOL(call_usermodehelper_freeinfo);
 
+static void umh_complete(struct subprocess_info *sub_info)
+{
+	complete(sub_info->complete);
+}
+
 /* Keventd can't block, but this (a child) can. */
 static int wait_for_helper(void *data)
 {
@@ -242,7 +247,7 @@ static int wait_for_helper(void *data)
 	if (sub_info->wait == UMH_NO_WAIT)
 		call_usermodehelper_freeinfo(sub_info);
 	else
-		complete(sub_info->complete);
+		umh_complete(sub_info);
 	return 0;
 }
 
@@ -277,7 +282,7 @@ static void __call_usermodehelper(struct work_struct *work)
 		/* FALLTHROUGH */
 
 	case UMH_WAIT_EXEC:
-		complete(sub_info->complete);
+		umh_complete(sub_info);
 	}
 }
 
